app/src/main/java/com/github/onlynotesswent/ui/common/Popups.kt

```diff
@@ -153,44 +161,89 @@ fun CreationDialog(
             }
       })
 }
+
 /**
- * Displays a popup dialog to browse and navigate the file system.
- *
- * This popup allows users to view folders in a hierarchical structure. Users can navigate into
- * subfolders, view folder contents, and return to the root folder.
+ * A composable function that displays a popup dialog for moving a specific item in the file system.
  *
- * @param onDismiss Callback invoked when the popup is dismissed.
- * @param folderViewModel The ViewModel that provides the folder data.
+ * @param onDismiss A callback triggered to dismiss the popup dialog.
+ * @param folderViewModel The ViewModel managing folder-related data and operations.
+ * @param onMoveHere A callback triggered when the "Move Here" button is clicked, designed to move
+ *   the selected item in a specific folder.
  */
 @Composable
-fun FileSystemPopup(onDismiss: () -> Unit, folderViewModel: FolderViewModel) {
+fun FileSystemPopup(
+    onDismiss: () -> Unit,
+    folderViewModel: FolderViewModel,
+    onMoveHere: (Folder?) -> Unit = {}
+) {
   var selectedFolder by remember { mutableStateOf<Folder?>(folderViewModel.selectedFolder.value) }
   var folderSubFolders by remember { mutableStateOf<List<Folder>>(emptyList()) }
   val userRootFolders = folderViewModel.userRootFolders.collectAsState()
 
+  // Modify the subfolder when selected Folder changes, best way I found how to do it as when done
+  // sequentially it takes a bit of time for the selected Folder to change which causes a bug
+  // where the subfolders don't update. Could fix this problem using a wait but that would depend
+  // on the internet speed of the user.
+  LaunchedEffect(selectedFolder) {
+    if (selectedFolder != null) {
+      folderViewModel.getSubFoldersOfNoStateUpdate(
```

Comment:
Maybe it is more appropriate to rename this function getSubFoldersOfFolderIdNoStateUpdate to make it a bit more specific

Reply:
Yes you're right I did it.